src/Universum/Container/Utils.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE Trustworthy #-}
+
+-- | Utility functions on top of 'Container' typeclass.
+
+module Universum.Container.Utils
+       ( groupByFst
+       , groupByKey
+       , groupByKeyBy
+
+       , groupOn
+       ) where
+
+import Data.Function (id, on, (.))
+import Data.List.NonEmpty (NonEmpty (..), groupBy)
+
+import Universum.Base
+import Universum.Container.Class
+
+-- $setup
+-- >>> import Data.Function
+-- >>> import GHC.Num
+-- >>> import Data.Text (toLower)
+
+-- | Variation of 'groupByKey' that accepts the comparison function on keys
+-- explicitly. This function must define a (computable) equivalence relation.
+--
+-- Among multiple keys appearing in the same group it will pick the leftmost
+-- one as the representer of the equivalence class.
+--
+-- >>> groupByKeyBy (<=) id [(1, 10), (2, 11), (0, 12), (1, 13), (3, 14)]
+-- [(1,10 :| [11]),(0,12 :| [13,14])]
```

Comment:
The documentation indicates that the function must define an equivalence relation, but `(<=)` does not do so! This example is busted (i.e., its result is unspecified).